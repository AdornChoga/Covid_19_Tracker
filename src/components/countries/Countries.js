import React from 'react';
import { useSelector } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';
import Country from './Country';
import classes from './Countries.module.css';

const Countries = () => {
  const { statistics } = useSelector((state) => state);
  return (
    <div className={classes.countries_container}>
      {
        statistics.countries.map((country) => (
          <Country country={country} key={uuidv4()} />
        ))
      }
    </div>
  );
};

export default Countries;
