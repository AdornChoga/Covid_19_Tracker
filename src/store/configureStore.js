import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import statistics from './statistics/statistics';

const reducer = combineReducers({
  statistics,
});

const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
