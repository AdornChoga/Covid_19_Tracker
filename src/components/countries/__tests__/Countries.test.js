import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../../store/configureStore';
import Countries from '../Countries';

it('should create countries component snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Countries />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
