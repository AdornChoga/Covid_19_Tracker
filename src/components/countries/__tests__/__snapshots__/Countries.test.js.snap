// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create countries component snapshot 1`] = `
<div
  className="countries_container"
/>
`;
